src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -169,7 +171,19 @@ public int order() {
                 attachSourceFieldContext(request);
             }
             final Set<String> injectedRoles = rolesInjector.injectUserAndRoles(threadContext);
+            boolean isInjectedUser = false;
+            if(threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER) == null
```

Comment:
There is call to `threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER)`, please reuse `User user`.